internal/cluster/distribution/eks/eksprovider/workflow/wait_update_eks_version_activity.go

```diff
@@ -0,0 +1,85 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+	"time"
+
+	"emperror.dev/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/eks"
+)
+
+const WaitUpdateClusterVersionActivityName = "eks-wait-update-version"
+
+// WaitUpdateClusterVersionActivity responsible for updating an EKS cluster
+type WaitUpdateClusterVersionActivity struct {
+	awsSessionFactory AWSFactory
+}
+
+// WaitUpdateClusterVersionActivityInput holds data needed for waiting for an EKS cluster update
+type WaitUpdateClusterVersionActivityInput struct {
+	EKSActivityInput
+
+	UpdateID string
+}
+
+// NewWaitUpdateClusterVersionActivity instantiates a new EKS version update waiting activity
+func NewWaitUpdateClusterVersionActivity(awsSessionFactory AWSFactory) *WaitUpdateClusterVersionActivity {
+	return &WaitUpdateClusterVersionActivity{
+		awsSessionFactory: awsSessionFactory,
+	}
+}
+
+func (a *WaitUpdateClusterVersionActivity) Execute(ctx context.Context, input WaitUpdateClusterVersionActivityInput) error {
+	session, err := a.awsSessionFactory.New(input.OrganizationID, input.SecretID, input.Region)
+	if err = errors.WrapIf(err, "failed to create AWS session"); err != nil {
+		return err
+	}
+
+	eksClient := eks.New(session)
+
+	ticker := time.NewTicker(30 * time.Second)
+	defer ticker.Stop()
```

Comment:
**Note**: I'm not sure if this ticker based waiting is the more conventional choice here, or the AWS status based request waiter solution applied [here](https://github.com/banzaicloud/pipeline/blob/master/internal/cluster/distribution/eks/eksprovider/workflow/create_eks_control_plane_activity.go#L207). Whichever is the current conventional approach (preferably the one containing less custom logic) is fine by me. Also apologies if I missed some condition which invalidates the other approach.

(The other approach bypasses the separate waiter activity entirely if I'm reading it correctly which seems to spare a lot of lines of code, but I really don't want to force it if the ticker is the better choice for our current conventions.)

Reply:
TBH I started to use the waiter to resolve this (I have never used it before), but doesn't gave me better looking or even shorter code inside `WaitUpdateClusterVersionActivity.Execute`.

Reply:
All right.
I'm leaving the discussion unresolved so others can read it, but otherwise I consider this thread closed.